Mutagen.Bethesda.Analyzers.SDK/Analyzers/IAnalyzer.cs

```diff
@@ -4,5 +4,6 @@ namespace Mutagen.Bethesda.Analyzers.SDK.Analyzers;
 
 public interface IAnalyzer
 {
+    AnalyzerId Id => new(GetType().FullName ?? throw new InvalidOperationException("Analyzer type has no FullName"));
```

Comment:
This being used for anything atm?

Reply:
I think it would need to be used when we want analyzer specific settings. Didn't have any of those yet though I'm realizing. Can you think of any applicable settings?

Reply:
I think we should focus on "topics" for the end user's API.  I think what analyzer is handling what topic and how is an implementation detail that shouldn't be exposed too much.    Oh.. topic ABC catches bug X.   It used to be fished for and reported via analyzer G, but now H does it as it's already doing this other thing and it's super optimized to also check X.

We dont want users to know about G or H to where the refactor would cause problems, i think?